Assets/Huawei/Demos/ML/LanguageDetection/LanguageDetectionDemoManager.cs

```diff
@@ -0,0 +1,188 @@
+    private void FirstBestDetectOnline()
+    {
+        Debug.Log("FirstBestDetectOnline");
+        Debug.Log("FirstBestDetectOnline" + langInputField.text);
+        HMSMLLanguageDetectionManager.Instance.FirstBestDetectRemote(langInputField.text, OnDetectString, OnDetectFailure);
+    }
+    private void SyncFirstBestDetectOnline()
+    {
+        Debug.Log("SyncFirstBestDetectOnline");
+        Debug.Log("SyncFirstBestDetectOnline" + langInputField.text);
+        var detectedText = HMSMLLanguageDetectionManager.Instance.SyncFirstBestDetectRemote(langInputField.text);
+        resultText.text = detectedText;
+    }
+    private void ProbilityDetectOnline()
+    {
+        Debug.Log("ProbilityDetectOnline");
+        Debug.Log("ProbilityDetectOnline" + langInputField.text);
+        HMSMLLanguageDetectionManager.Instance.ProbilityDetectRemote(langInputField.text, OnDetectList, OnDetectFailure);
+    }
+    private void SyncProbilityDetectOnline()
+    {
+        Debug.Log("SyncProbilityDetectOnline");
+        Debug.Log("SyncProbilityDetectOnline" + langInputField.text);
+        var detectedList = HMSMLLanguageDetectionManager.Instance.SyncProbilityDetectRemote(langInputField.text);
+        OnDetectList(detectedList);
+    }
+
+
+    //ML Local Lang Detector Functions (Offline)
+    private void FirstBestDetectOffline()
+    {
+        Debug.Log("FirstBestDetectOffline");
+        Debug.Log("FirstBestDetectOffline" + langInputField.text);
+        HMSMLLanguageDetectionManager.Instance.FirstBestDetectLocal(langInputField.text, OnDetectString, OnDetectFailure);
+    }
+    private void SyncFirstBestDetectOffline()
+    {
+        Debug.Log("SyncFirstBestDetectOffline");
+        Debug.Log("SyncFirstBestDetectOffline" + langInputField.text);
+        var detectedText = HMSMLLanguageDetectionManager.Instance.SyncFirstBestDetectLocal(langInputField.text);
+        resultText.text = detectedText;
+    }
+    private void ProbilityDetectOffline()
+    {
+        Debug.Log("ProbilityDetectOffline");
+        Debug.Log("ProbilityDetectOffline" + langInputField.text);
+        HMSMLLanguageDetectionManager.Instance.ProbilityDetectLocal(langInputField.text, OnDetectList, OnDetectFailure);
+    }
+
+    private void SyncProbilityDetectOffline()
+    {
+        Debug.Log("SyncProbilityDetectOffline");
+        Debug.Log("SyncProbilityDetectOffline" + langInputField.text);
+        var detectedList = HMSMLLanguageDetectionManager.Instance.SyncProbilityDetectLocal(langInputField.text);
+        OnDetectList(detectedList);
+    }
+
+
+
+
+    private void StopDetection() {
+
+        HMSMLLanguageDetectionManager.Instance.StopLocalLangDetector();
+        HMSMLLanguageDetectionManager.Instance.StopRemoteLangDetector();
+    }
+
+
+    private void OnDetectList(IList<MLDetectedLang> result) {
+        Debug.Log($"OnDetectList List Count: {result.Count}");
+
+        resultText.text = "";
+        if (result.Count!=0||result!=null)
```

Comment:
You can write as a more readable like below;

` if (result?.Count > 0)`


? mean is null check